### PR TITLE
Add search results highlighting to `ContentScopeSelect`

### PR DESCRIPTION
**Pull request by fichtnerma**
---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-1028
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details>
    <summary>Screenshots/screencasts</summary>
    
image

</details>

### CODE REVIEW

File: .changeset/cool-spoons-shave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-admin": minor
+---
+
+Add Search Highlighting for ContentScope Select
```
Comment (johnnyomair) [inline .changeset/cool-spoons-shave.md:5]:
```suggestion
Add search results highlighting to `ContentScopeSelect`
```

**Review by johnnyomair — COMMENTED** [review]
With your solution it's not easily possible to add the highlighting when using the `renderOption` option. I think we still should support that. @thomasdax98 what do you think?

File: storybook/src/cms-admin/ContentScopeSelect.tsx [hunk 2]
```diff
@@ -133,6 +133,32 @@ storiesOf("@comet/cms-admin/Content Scope Select", module)
...
+                    { domain: { label: "Main", value: "main" }, language: { label: "German", value: "de" } },
+                    { domain: { label: "Secondary", value: "secondary" }, language: { label: "English", value: "en" } },
+                    { domain: { label: "Secondary", value: "secondary" }, language: { label: "German", value: "de" } },
+                ]}
+                renderOption={(option, addSearchHighlighting) => (
+                    <>
+                        <ListItemIcon>
+                            <Domain />
+                        </ListItemIcon>
+                        <ListItemText primary={<>{addSearchHighlighting(`${option.domain.label} – ${option.language.label}`)}</>} />
```
Comment (johnnyomair) [inline storybook/src/cms-admin/ContentScopeSelect.tsx:156]: I'm not sure about this. IMO a better approach would be:

```ts
renderOption={(option, { query /* = searchValue */ }) => {
  const text = `${option.domain.label} – ${option.language.label}`
  const matches = findTextMatches(text, query);
  return (
    <ListItemText primary={<MarkedMatches text={text} matches={matches} />} />
  )
}
```

@nsams do you have a better idea to enable search results highlighting in a custom renderOption?

Comment (nsams) [reply storybook/src/cms-admin/ContentScopeSelect.tsx:156]: :+1: your idea

File: packages/admin/cms-admin/src/index.ts [hunk 3]
```diff
@@ -29,6 +29,7 @@ export { Header } from "./common/header/Header";
 export { UserHeaderItem } from "./common/header/UserHeaderItem";
 export type { TextMatch } from "./common/MarkedMatches";
 export { MarkedMatches } from "./common/MarkedMatches";
+export { findTextMatches } from "./common/MarkedMatches";
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/index.ts:32]: Please add a changeset for this.

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**